roofit/roofitcore/src/FitHelpers.cxx

```diff
@@ -66,7 +67,19 @@ constexpr int extendedFitDefault = 2;
 /// \param[in] data The dataset that was used for the fit.
 int calcAsymptoticCorrectedCovariance(RooAbsReal &pdf, RooMinimizer &minimizer, RooAbsData const &data)
 {
-   // Calculated corrected errors for weighted likelihood fits
+  RooFormulaVar logpdf("logpdf", "-log(pdf)", "-log(@0)", pdf);
+  RooArgSet obs;
+  logpdf.getObservables(data.get(), obs);
+
+  //warning if the dataset is binned
+  if (pdf.isBinnedDistribution(obs)) {
```

Comment:
```suggestion
  // Warning if the dataset is binned. TODO: in some cases,
  // people also use RooDataSet to encode binned data,
  // e.g. for simultaneous fits. It would be useful to detect
  // this in this future as well.
  if (dynamic_cast<RooDataHist const*>(&data)) {
```
...or something like that. In any case, we should not use `pdf.isBinnedDistribution(obs)` for this check, because one might also use binned distributions in an unbinned fit, and then this gives a false warning.